papaya-core/src/main/java/pl/edu/agh/papaya/service/sprint/SprintService.java

```diff
@@ -94,8 +88,23 @@ Sprint createSprint(Sprint sprint) {
         return sprintRepository.findFirstByProjectIdOrderByDurationPeriodStartDesc(projectId);
     }
 
-    public Sprint closeSprint(Sprint sprint, Duration timeBurned, Duration estimatedTimePlanned,
-            Duration finalTimePlanned, LocalDateTime dateClosed) {
+    public Sprint closeSprint(Sprint sprint, Long timeBurnedInMinutes, Long estimatedTimePlannedInMinutes,
```

Comment:
Czemu tutaj nie przekazujemy `Duration`?

Reply:
Bo potem następuje sprawdzenie, czy wartości są >= 0. Oryginalnie to było w `SprintsRestService` i rzucało tam `IllegalStateException`, ale przeniosłam to do `SprintService`. Niby można bezkrytycznie robić z tego `Duration` w `SprintsRestService`, a potem w  `SprintService` wyciągać minuty i sprawdzać, czy są nieujemne, ale nie wydawało mi się to super ładne, a nie chciało mi się przerabiać jakoś bardzo utilsowej metody i generalnie nieswojego kodu

Reply:
[`Duration.isNegative`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#isNegative--)

Poza tym co było złego w sprawdzaniu tego w SprintsRestService? Walidujemy dane od użytkownika

Reply:
Znaczy głównie chodziło mi o to, że `IllegalStateException` leciało w paru miejscach i trafiało na front. Przy przerabianiu liczb na `Duration` jeszcze by trzeba osobno sprawdzić, czy liczba nie jest nullem i chyba dlatego powstała gdzieś tam osobna metoda, która robi obie te rzeczy. Ale spoko, mogę to przywrócić do `SprintsRestService`, po prostu akurat tak mi się to przetasowało

Reply:
Teraz też musisz sprawdzać czy `Long` nie jest nullem.

Chodzi głównie o to, że przez te longi interfejs jest taki sobie. Według mnie `closeSprint(Sprint, Duration, ...)` wygląda lepiej i jest łatwiejszy w użyciu niż `closeSprint(Sprint, Long, ...)`, który i tak nie powinien akceptować `Long` tylko `long` bo nie chcemy nulli.

Reply:
> Teraz też musisz sprawdzać czy `Long` nie jest nullem.

No wiem, jest na to metodka, która równocześnie sprawdza, czy jest >= 0 - tak to było robione wcześniej, ja tylko przeniosłam wyjątkogenne rzeczy w jedno miejsce

 
> Chodzi głównie o to, że przez te longi interfejs jest taki sobie. Według mnie `closeSprint(Sprint, Duration, ...)` wygląda lepiej i jest łatwiejszy w użyciu niż `closeSprint(Sprint, Long, ...)`, który i tak nie powinien akceptować `Long` tylko `long` bo nie chcemy nulli.

No pewnie, ma to sens, nie jestem przywiązana do tego układu - powstał w trakcie ~losowego~ przeklejania kawałków kodu w inne miejsca, także absolutnie się nie kłócę, pokornie poprawię